### PR TITLE
Resolve attack and thrust orders

### LINKED ISSUE #44: Resolve thrust command

### LINKED ISSUE #45: Resolve attack command
Might need to do this in parallel - a ship that just got it's guns disabled can still attack

### PR DESCRIPTION
**Pull request by JustinHuPrime**
Not unit tested

Closes #44 
Closes #45

### CODE REVIEW

File: src/common/Board.hs [hunk 1]
```diff
@@ -15,18 +15,32 @@ PARTICULAR PURPOSE. See the GNU General Public License for more details.
 You should have received a copy of the GNU Affero General Public License along
 with N-Planetary. If not, see <https://www.gnu.org/licenses/>.
 -}
+{-# LANGUAGE NamedFieldPuns #-}
```
Comment (wyndro) [inline src/common/Board.hs:18]: Huh?

File: src/common/Board.hs [hunk 2]
```diff
@@ -15,18 +15,32 @@ PARTICULAR PURPOSE. See the GNU General Public License for more details.
...
+import Serializing
 import Util
 
 type Board = [Entity]
 
 data Entity
   = --- Astronomical object with some id, some x, y position, some name, some mass, some radius (in natural units)
-    AstroObj Int Vec2 String Double Double
+    AstroObj
+      { idNum :: Int,
```
Comment (wyndro) [inline src/common/Board.hs:33]: Oooo, fun

File: src/common/Board.hs [hunk 3]
```diff
@@ -37,8 +51,129 @@ data Entity
...
+
+findId :: Int -> Board -> Maybe Entity
+findId searchFor = find (\e -> searchFor == idNum e)
+
+isShip :: Entity -> Bool
+isShip Ship {} = True
+isShip _ = False
+
+doDamage :: (Int, Int, Int) -> Entity -> Entity
+doDamage (toWeapon, toDrive, toStructure) ship@Ship {weaponDamage, driveDamage, structureDamage} =
```
Comment (wyndro) [inline src/common/Board.hs:81]: What do the curly braces do?

Comment (wyndro) [reply src/common/Board.hs:81]: What is this fancy thing you have done?

File: src/test/TestMain.hs [hunk 4]
```diff
@@ -16,12 +16,28 @@ You should have received a copy of the GNU Affero General Public License along
...
+module Main where
+
+import Test.Framework
+import qualified TestBoard
+import qualified TestFilterVisible
+import qualified TestMove
+import qualified TestOrderValidator
+
+tests :: [Test]
+tests = [TestBoard.group, TestMove.group, TestOrderValidator.group, TestFilterVisible.group]
```
Comment (wyndro) [inline src/test/TestMain.hs:28]: Is there a test for order resolving? Will there be?

File: src/common/Board.hs (hunk 1 above)
Comment (JustinHuPrime) [reply src/common/Board.hs:18]: It's a thingy that makes brace-notation easier

File: src/test/TestMain.hs (hunk 4 above)
Comment (JustinHuPrime) [reply src/test/TestMain.hs:28]: See issue #55 

I will add test eventually

File: src/common/Board.hs (hunk 3 above)
Comment (JustinHuPrime) [reply src/common/Board.hs:81]: This is record notation - the weaponDamage field gets bound to the identifier weaponDamage (that's why I need the pun language option)

The at-pattern says that if the argument in question matches the following pattern, bind the whole operand to the name ship.

**Review by JohannCooper — APPROVED** [review]

**Review by wyndro — APPROVED** [review]

**Pull request merged**